src/metabase/query_processor/middleware/visualization_settings.clj

```diff
@@ -44,10 +44,11 @@
 (defn- viz-settings
   "Pull viz settings from either the query map or the DB"
   [query]
-  (or (let [viz (-> query :viz-settings)]
-        (when (seq viz) viz))
-      (when-let [card-id (-> query :info :card-id)]
-        (mb.viz/db->norm (:visualization-settings (lib.metadata.protocols/card (qp.store/metadata-provider) card-id))))))
+  (mb.viz/db->norm
+   (or (let [viz (-> query :viz-settings)]
+         (when (seq viz) viz))
```

Comment:
it's not your code, but it could say `(-> query :viz-settings not-empty)` to be a bit easier to read.

Reply:
Hmm.. I think this is actually to cover the case where :viz-settings IS there, but it's just an empty map, hence the `(seq viz)` check.

Reply:
`not-empty` does exactly that :) `(not-empty {}) ; => nil`